File: Armstrong_or_not.c
#include
#include
int main()
{
int number, temp, remainder, result = 0, n = 0 ;
printf(“Enter an integer: “);
scanf(“%d”, &number);
temp = number;
while (temp != 0)
{
temp /= 10;
++n;
}
temp = number;
while (temp != 0)
{
remainder = temp%10;
result += pow(remainder, n);
temp /= 10;
}
if(result == number)
printf(“%d is an Armstrong number\n”, number);
else
printf(“%d is not an Armstrong number\n”, number);
return 0;
}

File: Armstrong_or_not.java
package trainingday5;
import java.util.*;
public class Armstrong {
	public static void main(String[] args)
{   Scanner sc=new Scanner(System.in);
	int n=0,temp,rem,res=0,number;
	System.out.println("enter a number ");
	number=sc.nextInt();
	temp=number;
	while(temp!=0)
	{
		temp/=10;
		++n;
	}
	temp= number;
	while(temp!=0)
	{
		rem=temp%10;
		res+=Math.pow(rem,n);
		temp/=10;
	}
	if(res==number)
	{
		System.out.println("Armstrong ");
	}
	else 
		System.out.println("not Armstrong ");
}
}
